src/components/theharvester/theharvester.tsx

```diff
@@ -1,87 +1,121 @@
-    { value: "rapiddns", label: "RapidDNS" },
-    { value: "sitedossier", label: "Sitedossier" },
-    { value: "subdomaincenter", label: "Subdomaincenter" },
-    { value: "subdomainfinderc99", label: "Subdomainfinderc99" },
-    { value: "threatminer", label: "Threatminer" },
-    { value: "urlscan", label: "URLScan" },
-    { value: "virustotal", label: "Virustotal" },
-    { value: "yahoo", label: "Yahoo" },
-];
 
+/**
+ * The Harvester component.
+ * @returns The Harvester component.
+ */
 const TheHarvester = () => {
-    const [loading, setLoading] = useState(false);
-    const [output, setOutput] = useState("");
-    const [checkedAdvanced, setCheckedAdvanced] = useState(false);
-    const [selectedSource, setSelectedSource] = useState("");
-    const [pid, setPid] = useState("");
+    // Component State Variables.
+    const [loading, setLoading] = useState(false); // State variable to indicate loading state.
+    const [output, setOutput] = useState(""); // State variable to store the output of the command execution.
+    const [checkedAdvanced, setCheckedAdvanced] = useState(false); // State variable to indicate advanced settings.
+    const [pid, setPid] = useState(""); // State variable to store the process ID of the command execution.
+    const [isCommandAvailable, setIsCommandAvailable] = useState(false); // State variable to check if the command is available.
+    const [opened, setOpened] = useState(!isCommandAvailable); // State variable that indicates if the modal is opened.
+    const [loadingModal, setLoadingModal] = useState(true); // State variable to indicate loading state of the modal.
+    const [allowSave, setAllowSave] = useState(false);
+    const [hasSaved, setHasSaved] = useState(false);
+
+    // Component Constants.
+    const title = "The Harvester"; // Title of the component.
+    const description =
+        "A tool for gathering subdomain names, e-mail addresses, virtual hosts, open ports/ banners, and employee names from different public sources (search engines, pgp key servers)."; // Description of the component.
+    const steps =
+        "Step 1: Enter a valid domain to be harvested.\n" +
+        "       Eg: kali.org\n" +
+        "Step 2: Enter a limit for the requests. Default is 500 results. Can be left blank.\n" +
+        "       Eg: 500\n" +
+        "Step 3: Select a source to search form. The list contains compatible search engines.\n" +
+        "       Eg: baidu\n" +
+        "Step 4: Click Start The Harvester to commence the tool's operation.\n" +
+        "Step 5: View the Output block below to view the results of the tool's execution.\n" +
+        "Switch to Advanced Mode for further options.";
+    const sourceLink = "https://gitlab.com/kalilinux/packages/theharvester"; // Link to the source code.
+    const tutorial = ""; // Link to the official documentation/tutorial.
+    const dependencies = ["TheHarvester"]; // Contains the dependencies required by the component.
```

Comment:
Dependency needs to match the application that gets installed. 
```suggestion
    const dependencies = ["theHarvester"]; // Contains the dependencies required by the component.
```